### DISCUSSION TITLE
Blazor rendering modes

### DISCUSSION THREAD

**Discussion by Rodien** [oqtane/oqtane.framework#3433] [General]
I just watched this video about rendering modes.

I'm posting it here for anyone who finds the rendering modes a bit confusing.

https://youtu.be/YUp7JJI4pqg

**Comment by sbwalker** [comment]
The confusing part for me is that the apparent problem which Microsoft was trying to solve in .NET 8 is "not forcing developers to have to decide to use Blazor Server or Blazor WebAssembly" when they start a Blazor project. 

As we all know, Oqtane has supported Blazor Server AND WebAssembly in the same application/installation since Blazor was introduced - so this was never an actual problem - it was more that the default Blazor templates from Microsoft did not showcase that this was possible. 

In addition, as we all know in the Oqtane community, the hosting model (now referred to as render modes) is something that can NOT be easily ignored - as in order to support both hosting models you must architect and develop your application using a SPA (client/server) approach - where you have distinct and separate client and server applications that communicate via HTTP. 

So in order to use the Blazor multi-target capabilities in .NET 8 you need to follow the development approach which Oqtane has used all along... however this approach was not widely understood by the majority of Blazor Server developers , so they will face a significant migration effort if they want to use the new capabilities in .NET 8.

**Comment by Rodien** [comment]
If I understand it correctly, there are two new modes coming to .NET 8 that Oqtane does not support, and we don't know if Oqtane will support them. These modes are SSR (think MVC/Razor Pages) and 'Interactive Auto' which runs as follows: 'Server, then client.'

I must say that I like the prospect of a pure SSR Blazor application. You can add features like 'stream rendering' or use something like HTMX to add interactivity.



> The confusing part for me is that the apparent problem which Microsoft was trying to solve in .NET 8 is "not forcing developers to have to decide to use Blazor Server or Blazor WebAssembly" when they start a Blazor project.'

I think this why they introduce 'Interactive Auto'.
